core/changedetector/proto_change_detector_service.go

```diff
@@ -0,0 +1,215 @@
+	for _, newEnumDesc := range fds.GetEnumType() {
+		enumName := fds.GetPackage() + "." + newEnumDesc.GetName()
+		if oldEnumDesc := getEnumDescriptor(packageEnumMap, fds.GetPackage(), newEnumDesc.GetName()); oldEnumDesc != nil {
+			if !proto.Equal(newEnumDesc, oldEnumDesc) {
+				sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+				appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+			}
+		} else {
+			sce.UpdatedSchemas = append(sce.UpdatedSchemas, enumName)
+			appendImpactedFields(sce, enumName, getImpactedEnumFields(oldEnumDesc, newEnumDesc))
+		}
+	}
+}
+
+func getPackageMessageMap(fileDescriptorSet *descriptor.FileDescriptorSet) map[string]map[string]*descriptor.DescriptorProto {
+	packageMessageMap := make(map[string]map[string]*descriptor.DescriptorProto)
+	for _, fileDescriptor := range fileDescriptorSet.GetFile() {
+		pkgName := fileDescriptor.GetPackage()
+		if _, ok := packageMessageMap[pkgName]; !ok {
+			packageMessageMap[pkgName] = make(map[string]*descriptor.DescriptorProto)
+		}
+		for _, messageDescriptor := range fileDescriptor.GetMessageType() {
+			packageMessageMap[pkgName][messageDescriptor.GetName()] = messageDescriptor
+		}
+	}
+	return packageMessageMap
+}
+
+func getPackageEnumMap(fileDescriptorSet *descriptor.FileDescriptorSet) map[string]map[string]*descriptor.EnumDescriptorProto {
+	packageMessageMap := make(map[string]map[string]*descriptor.EnumDescriptorProto)
+	for _, fileDescriptor := range fileDescriptorSet.GetFile() {
+		pkgName := fileDescriptor.GetPackage()
+		if _, ok := packageMessageMap[pkgName]; !ok {
+			packageMessageMap[pkgName] = make(map[string]*descriptor.EnumDescriptorProto)
+		}
+		for _, enumDescriptor := range fileDescriptor.GetEnumType() {
+			packageMessageMap[pkgName][enumDescriptor.GetName()] = enumDescriptor
+		}
+	}
+	return packageMessageMap
+}
+
+func getMessageDescriptor(packageMessageMap map[string]map[string]*descriptor.DescriptorProto, packageName, messageName string) *descriptor.DescriptorProto {
+	if packageMap, found := packageMessageMap[packageName]; found {
+		if descriptor, found := packageMap[messageName]; found {
+			return descriptor
+		}
+	}
+	return nil
+}
+
+func getEnumDescriptor(packageEnumMap map[string]map[string]*descriptor.EnumDescriptorProto, packageName, enumName string) *descriptor.EnumDescriptorProto {
+	if packageMap, found := packageEnumMap[packageName]; found {
+		if descriptor, found := packageMap[enumName]; found {
+			return descriptor
+		}
+	}
+	return nil
+}
+
+func findEnumDescriptorFromMessageDescriptor(messageDescriptor *descriptor.DescriptorProto, enumName string) *descriptor.EnumDescriptorProto {
+	for _, enumDescriptor := range messageDescriptor.GetEnumType() {
+		if enumDescriptor.GetName() == enumName {
+			return enumDescriptor
+		}
+	}
+	return nil
+}
+
+func populateReverseDependencies(fileDescriptorSet *descriptor.FileDescriptorSet, reverseDependencies map[string][]string) {
```

Comment:
Same here, Instead of populate use get approach here, pass the result as return value and zero value in case if you expect zero value.

Reply:
updated